frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1235,6 +1235,23 @@ export const surveyLogic = kea<surveyLogicType>([
                 return null
             },
         ],
+        deviceTypesMatchTypeValidationError: [
+            (s) => [s.survey],
+            (survey: Survey): string | null => {
+                if (
+                    survey.conditions?.deviceTypesMatchType === SurveyMatchType.Regex &&
```

Comment:
there are 2 types, matches regex and does not match regex, so we should validate its a valid regex in both of them